controllers/update_status.go

```diff
@@ -131,50 +142,34 @@ func (s UpdateStatus) Reconcile(r *FoundationDBClusterReconciler, context ctx.Co
 		processClass := instance.GetProcessClass()
 		instanceID := instance.GetInstanceID()
 
+		// Even the instance will be removed we need to keep the config around
+		if processClass == fdbtypes.ProcessClassStorage {
+			status.AddStorageServerPerDisk(getStorageServersPerPodForInstance(&instance))
+		}
+
 		if cluster.InstanceIsBeingRemoved(instanceID) {
 			continue
 		}
 
 		status.ProcessCounts.IncreaseCount(processClass, 1)
 
-		processStatus := processMap[instanceID]
-		if len(processStatus) == 0 {
-			existingTime, exists := cluster.Status.MissingProcesses[instanceID]
-			if exists {
-				status.MissingProcesses[instanceID] = existingTime
-			} else {
-				status.MissingProcesses[instanceID] = time.Now().Unix()
+		if processClass == fdbtypes.ProcessClassStorage && cluster.GetStorageServersPerPod() > 1 {
+			storageServersPerPod := cluster.GetStorageServersPerPod()
```

Comment:
Do we want to use the cluster-level setting here, or the instance-level setting?

Reply:
Good point 👍